subprojects/docs/src/snippets/kotlinDsl/assignment/kotlin/build.gradle.kts

```
@@ -0,0 +1,34 @@
+plugins {
+    `base`
+}
+
+// tag::assignment[]
+interface MyExtension {
+    val input: Property<String>
+    val output: RegularFileProperty
+}
+
+abstract class MyTask : DefaultTask() {
+    @get:Input
+    abstract val input: Property<String>
+    @get:OutputFile
+    abstract val output: RegularFileProperty
+
+    @TaskAction
+    fun execute() {
+        output.get().asFile.writeText(input.get())
+    }
+}
+
+val extension = extensions.create<MyExtension>("myExtension").apply {
```

I think we should use the same name for the variable and the registered extension:

```suggestion
val extension = extensions.create<MyExtension>("extension").apply {
```